crates/core_arch/src/simd.rs

```diff
@@ -3,25 +3,22 @@
 #![allow(non_camel_case_types)]
 
 macro_rules! simd_ty {
-    ($id:ident [$ety:ident]: $($elem_ty:ident),* | $($elem_name:ident),*) => {
+    ($id:ident: [$ety:ident; $ecount:literal]) => {
         #[repr(simd)]
         #[derive(Copy, Clone, Debug, PartialEq)]
-        pub(crate) struct $id($(pub $elem_ty),*);
+        pub(crate) struct $id(pub [$ety; $ecount]);
 
         #[allow(clippy::use_self)]
         impl $id {
             #[inline(always)]
-            pub(crate) const fn new($($elem_name: $elem_ty),*) -> Self {
```

Comment:
That is less ergonomic, shall we consider adding `From` implementation to support both tuple and array and keep new as it is now?

Reply:
Note that this was always `pub(crate)`, so users couldn't call it anyway.

They can only be created using intrinsics (or memory stuff like transmutes) in stable today: <https://doc.rust-lang.org/core/arch/x86/struct.__m128i.html>.

(The portable-simd types *do* have nice safe public constructors.  But these arch ones intentionally don't.)